sqflite_common/lib/src/batch.dart

```diff
@@ -114,21 +114,21 @@ abstract class SqfliteBatch implements Batch {
 /// Batch on a given database
 class SqfliteDatabaseBatch extends SqfliteBatch {
   /// Create a batch in a database
-  SqfliteDatabaseBatch(this.database, this._startTransaction);
+  SqfliteDatabaseBatch(this.database);
 
   /// Our database
   final SqfliteDatabase database;
 
-  /// Whether this batch should be executed in a transaction, or whether we
-  /// assume that the user is responsible for managing the transaction for us.
-  final bool _startTransaction;
-
   @override
-  Future<List<Object?>> commit(
-      {bool? exclusive, bool? noResult, bool? continueOnError}) {
+  Future<List<Object?>> commit({
+    bool? startTransaction,
```

Comment:
Thanks !

and yes; I think I like it better to control at the action level. I remember wondering whether `commit` could be named `apply`...

Other arguments (`exclusive`, `noResult`, `continueOnError`) assume false for null while `startTransaction` assumes true. We have several choices here:
```dart
bool? startTransaction, // default meaning true
bool startTransaction = true, // i.e. should we define a default value explicitely
bool? startTransaction = true // i.e. could we allow null (good sometimes to avoid parameter forwarding),
bool? noTransaction, // default being false
bool noTransaction = false, // explicit default value
bool? noTransaction = false, // is this even done elsewhere...
```

We could also create a new method `apply` - if you think, the name `commit` looks too much toward using a transaction.

Your suggestion are welcome. Maybe that is not worth a discussion, I just wonder whether there are any standard and consistency to look at.

Thanks again for your work on that.

Reply:
We could even have:
```dart
required bool startTransaction, // ha ha breaking compatibility for nothing!
```
Sorry...

If you're fine with what you have done (and that looks very fine to me), I'll merge that in a couple of days. You make an awesome job in maintaining sqlite3 libs and all. I'm very grateful.

Reply:
> We could also create a new method `apply` - if you think, the name `commit` looks too much toward using a transaction.

I really like that idea! It also makes it impossible to misuse the APIs by combining `startTransaction: false` and `exclusive: true`. Nice call, I'll change the API to that.

Reply:
Thanks, I think you'll notice that I define the API in the abstract class and implements it in the common mixin so that it is shared by all implementation (including sqflite_cipher) so that every subclass benefits from it.